test/e2e/framework/kcp.go

```diff
@@ -343,24 +344,18 @@ func (c *kcpServer) loadCfg() error {
 	var lastError error
 	if err := wait.PollImmediateWithContext(c.ctx, 100*time.Millisecond, 1*time.Minute, func(ctx context.Context) (bool, error) {
 		c.kubeconfigPath = filepath.Join(c.dataDir, "admin.kubeconfig")
-		if fs, err := os.Stat(c.kubeconfigPath); os.IsNotExist(err) {
-			lastError = err
-			return false, nil
-		} else if err != nil {
-			lastError = fmt.Errorf("failed to read admin kubeconfig after kcp start: %w", err)
-			return false, nil
-		} else if fs.Size() == 0 {
-			return false, nil
-		}
-
-		rawConfig, err := clientcmd.LoadFromFile(c.kubeconfigPath)
+		config, err := loadKubeConfig(c.kubeconfigPath)
 		if err != nil {
-			lastError = fmt.Errorf("failed to load admin kubeconfig: %w", err)
+			// A missing file is likely caused by the server not
+			// having started up yet. Ignore these errors for the
+			// purposes of logging.
+			if !os.IsNotExist(err) {
```

Comment:
there is a race that can lead to other errors: zero file size or corrupted because partially written file.

Reply:
Isn't that still handled by loadKubeConfig? I was trying to maintain the existing semantics, just factor out the common bits for reuse.